test/screenshot/lib/image-cache.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('mz/fs');
+const fsx = require('fs-extra');
+const os = require('os');
+const path = require('path');
+const request = require('request-promise-native');
+
+/**
+ * Downloads image files from public URLs and saves them to a stable path in the TEMP directory for later retrieval.
+ */
+class ImageCache {
+  constructor() {
+    /**
+     * @type {string}
+     * @private
+     */
+    this.tempDirPath_ = path.join(os.tmpdir(), 'mdc-web-image-cache');
```

Comment:
What does `os.tmpdir()` do here?

Reply:
It returns the path of the OS-defined "temporary" directory:

- `/tmp` on many *nix distros
- `$TMPDIR` on macOS (seems to be a randomly-generated path)
- `%TEMP%` on Windows (points to `%USERPROFILE%\AppData\Local\Temp`)

My goal was to avoid downloading the same image files from `master`'s `golden.json` every time you run a diff.

Reply:
Nice!